src/sinks/appsignal/mod.rs

```diff
@@ -0,0 +1,265 @@
+        .sink_map_err(|error| error!(message = "Fatal AppSignal sink error.", %error));
+
+        let healthcheck = healthcheck(
+            endpoint_uri(self.endpoint.clone(), "vector/healthcheck")?,
+            push_api_key,
+            client,
+        )
+        .boxed();
+
+        Ok((super::VectorSink::from_event_sink(sink), healthcheck))
+    }
+
+    fn input(&self) -> Input {
+        Input::new(DataType::Metric | DataType::Log)
+    }
+
+    fn acknowledgements(&self) -> &AcknowledgementsConfig {
+        &self.acknowledgements
+    }
+}
+
+pub struct AppsignalEventEncoder {
+    transformer: Transformer,
+}
+
+impl HttpEventEncoder<serde_json::Value> for AppsignalEventEncoder {
+    fn encode_event(&mut self, mut event: Event) -> Option<serde_json::Value> {
+        self.transformer.transform(&mut event);
+
+        match event {
+            Event::Log(log) => Some(json!({ "log": log })),
+            Event::Metric(metric) => Some(json!({ "metric": metric })),
+            _ => panic!("The AppSignal sink does not support this event: {event:?}"),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl HttpSink for AppsignalSinkConfig {
+    type Input = serde_json::Value;
+    type Output = Vec<BoxedRawValue>;
+    type Encoder = AppsignalEventEncoder;
+
+    fn build_encoder(&self) -> Self::Encoder {
+        AppsignalEventEncoder {
+            transformer: self.encoding.clone(),
+        }
+    }
+
+    async fn build_request(&self, events: Self::Output) -> crate::Result<Request<Bytes>> {
+        let uri = endpoint_uri(self.endpoint.clone(), "vector/events")?;
+        let request = Request::post(uri)
+            .header(
+                AUTHORIZATION,
+                format!("Bearer {}", self.push_api_key.inner()),
+            )
+            .header("Content-Encoding", "gzip");
+
+        let mut body = crate::serde::json::to_bytes(&events).unwrap().freeze();
+
+        let buffer = BytesMut::new();
+        let mut writer = GzEncoder::new(buffer.writer(), flate2::Compression::new(6));
+        writer.write_all(&body).expect("Writing to Vec can't fail");
+        body = writer
+            .finish()
+            .expect("Writing to Vec can't fail")
+            .into_inner()
+            .into();
+
+        request.body(body).map_err(Into::into)
+    }
+}
+
+pub(crate) async fn healthcheck(
+    uri: Uri,
+    push_api_key: String,
+    client: HttpClient,
+) -> crate::Result<()> {
+    let request = Request::get(uri).header(AUTHORIZATION, format!("Bearer {}", push_api_key));
+    let response = client.send(request.body(Body::empty()).unwrap()).await?;
+
+    match response.status() {
+        status if status.is_success() => Ok(()),
+        other => Err(super::HealthcheckError::UnexpectedStatus { status: other }.into()),
+    }
+}
+
+fn endpoint_uri(endpoint: String, path: &str) -> crate::Result<Uri> {
+    let mut uri = endpoint;
+    if !uri.ends_with('/') {
+        uri.push('/');
+    }
+    uri.push_str(path);
```

Comment:
I think passing in the endpoint as a reference and allocating a string inside the function is cleaner, something like:

```suggestion
fn endpoint_uri(endpoint: &str, path: &str) -> crate::Result<Uri> {
    let uri = if endpoint.ends_with('/') {
        format!("{endpoint}{path}")
    else {
        format!("{endpoint}/{path}")
    };
```

Then you don't have to clone at the callsite.